5-WebApp-AuthZ/5-1-Roles/README-incremental-instructions.md

```diff
@@ -0,0 +1,315 @@
+
+- You can also use PowerShell scripts that **automatically** creates these two roles for your app. They additionally create two users in your tenant and assign them to these two roles. If you want to use this automation:
+
+  1. In PowerShell run the following command to create the two roles and users.
+
+     ```PowerShell
+     .\AppCreationScripts\CreateUsersAndRoles.ps1
+     ```
+
+  1. Run the following command to remove the two roles and users once you done testing.
+
+     ```PowerShell
+     .\AppCreationScripts\CleanupUsersAndRoles.ps1
+     ```
+
+> Remember to make changes in the following line (the app name) in the powershell scripts to ensure that the scripts target the app you intended to use.
+
+ ```PowerShell
+      $app = Get-AzureADApplication -Filter "DisplayName eq 'WebApp-RolesClaims'"
+ ```
+
+   > More details on how to run the scripts are described in [App Creation Scripts](./AppCreationScripts/AppCreationScripts.md)
+
+### Step 4: Run the sample
+
+1. Clean the solution, rebuild the solution, and run it.
+
+1. Open your web browser and make a request to the app. The app immediately attempts to authenticate you via the Microsoft identity platform endpoint. Sign in with a work or school account.
+
+1. You will be prompted to consent during the sign-in process.
+
+![First time Consent](ReadmeFiles/Sign-in-Consent.png)
+
+1. On the home page, the app lists the various claims it obtained from your [ID token](https://docs.microsoft.com/en-us/azure/active-directory/develop/id-tokens). You'd notice a claim named `roles`. There will be one `roles` claim for each app role the signed-in use is assigned to.
+
+1. There also are two links provided on the home page under the **Try one of the following Azure App Role driven operations** heading. These links will result in an access denied error if the signed-in user is not present in the expected role. Sign-out and sign-in with a user account with the correct role assignment to view the contents of these pages.
+
+> Note: You need to be a tenant admin to view the page that lists all the groups and roles the signed-in user is assigned to. It requires the **Directory.Read.All** permission to work. If you run into the **AADSTS65001: The user or administrator has not consented to use the application** error, provide [admin consent](https://docs.microsoft.com/en-us/azure/active-directory/manage-apps/configure-user-consent#grant-admin-consent-when-registering-an-app-in-the-azure-portal) to your app in the portal. Sign-out and sign-in again to make the page work as expected.
+
+When you click on the page that fetches the signed-in user's roles and group assignments, the sample will attempt to obtain consent from you for the **Directory.Read.All** permission using [incremental consent](https://docs.microsoft.com/en-us/azure/active-directory/develop/azure-ad-endpoint-comparison#incremental-and-dynamic-consent).
```

Comment:
I did not get this incremental consent. I got the following error instead:
```
MsalUiRequiredException: AADSTS65001: The user or administrator has not consented to use the application
```

Is incremental consent still part of this sample?